### PR TITLE
Add stack_data 0.2.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
License: https://github.com/alexmojaki/stack_data/blob/v0.2.0/LICENSE.txt
Requirements:
- https://github.com/alexmojaki/stack_data/blob/v0.2.0/pyproject.toml
- https://github.com/alexmojaki/stack_data/blob/v0.2.0/setup.cfg

Actions:
1. Add missing host dependencies: toml, wheel
2. Fix python in host
3. Update doc_url
4. Fix home and dev urls with HTTPS

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
```
conda.exceptions.ResolvePackageNotFound:
  - executing
  - asttokens
  - pure_eval
```
We need to add `executing`, `asttokens`, `pure_eval`

**Review by dholth — APPROVED** [review]

**Review by dholth — APPROVED** [review]

**Review by dholth — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,20 +11,22 @@ source:
...
   noarch: python
 
 requirements:
   host:
     - pip
-    - python >=3.5
+    - python
     - setuptools >=44
     - setuptools_scm >=3.4.3
+    - toml
```
Comment (dholth) [inline recipe/meta.yaml:23]: pip includes a vendored `tomli` library, the `toml` dependency may not be necessary. setuptools does not parse toml at all - except in a test that runs `import tomli as toml`

Comment (skupr-anaconda) [reply recipe/meta.yaml:23]: `toml` is mentioned in the upstream pyproject.toml https://github.com/alexmojaki/stack_data/blob/80ea27810df097dd844351266983d3516425901f/pyproject.toml#L2

Comment (dholth) [reply recipe/meta.yaml:23]: expands to a requirement for `setuptools >42` https://github.com/pypa/setuptools_scm/blob/main/setup.py#L62

which must be a PEP 517 supporting version

**Pull request merged**